GooglePay/src/main/java/com/braintreepayments/api/GooglePayClient.java

```diff
@@ -190,33 +199,41 @@ public void requestPayment(@NonNull final FragmentActivity activity, @NonNull fi
             return;
         }
 
-        braintreeClient.getConfiguration(new ConfigurationCallback() {
+        braintreeClient.getAuthorization(new AuthorizationCallback() {
             @Override
-            public void onResult(@Nullable Configuration configuration, @Nullable Exception e) {
-                if (configuration == null) {
-                    callback.onResult(e);
-                    return;
-                }
+            public void onAuthorizationResult(@Nullable final Authorization authorization, @Nullable Exception authError) {
+                if (authorization != null) {
+                    braintreeClient.getConfiguration(new ConfigurationCallback() {
+                        @Override
+                        public void onResult(@Nullable Configuration configuration, @Nullable Exception configError) {
+                            if (configuration == null) {
+                                callback.onResult(configError);
+                                return;
+                            }
+
+                            if (!configuration.isGooglePayEnabled()) {
+                                callback.onResult(new BraintreeException("Google Pay is not enabled for your Braintree account," +
+                                        " or Google Play Services are not configured correctly."));
+                                return;
+                            }
+
+                            setGooglePayRequestDefaults(configuration, authorization, request);
+                            braintreeClient.sendAnalyticsEvent("google-payment.started");
+
+                            PaymentDataRequest paymentDataRequest = PaymentDataRequest.fromJson(request.toJson());
+                            Intent intent = new Intent(activity, GooglePayActivity.class)
+                                    .putExtra(EXTRA_ENVIRONMENT, getGooglePayEnvironment(configuration))
+                                    .putExtra(EXTRA_PAYMENT_DATA_REQUEST, paymentDataRequest);
+
+                            activity.startActivityForResult(intent, BraintreeRequestCodes.GOOGLE_PAY);
```

Comment:
```suggestion
                            // TODO: Use activity result API to launch this activity
                            activity.startActivityForResult(intent, BraintreeRequestCodes.GOOGLE_PAY);
```

Just noticed maybe we should add a TODO to update this internal activity launching to the new API as well